pennylane/ops/functions/equal.py

```diff
@@ -99,3 +114,14 @@ def equal(
                 return False
 
     return getattr(op1, "inverse", False) == getattr(op2, "inverse", False)
+
+
+def equal_measurements(op1, op2):
+    """Determine whether two MeasurementProcess objects are equal"""
+    return_types_match = op1.return_type == op2.return_type
+    observables_match = op1.obs == op2.obs
+    wires_match = op1.wires == op2.wires
```

Comment:
Should we also check if the two `MeasurementProcess` objects have the same `eigvals` and `log_base`? 
Somewhat related question, is it worthwhile to define the __eq__ function for the `MeasurementProcess` class instead of this?

Reply:
Yes, better to check the `eigvals` and `log_base` attributes too

Reply:
And regarding the `__eq__` magic method, I believe we can't add it because classes that define this method are unhashable, and we use the hash in PennyLane

Reply:
There are certain things we need to discuss internally before deciding to use `__eq__` instead. 

I would just add another branch in the `qml.equal` function for now.

Reply:
Not sure what you mean exactly by adding a branch to `qml.equal` in this context. The comparison would happen in `equal_measurements` by adding `eigvals_match` and `log_base_match`.

Reply:
Adding a conditional branch means adding another `if` statement, which is what you already did